lib/ng2-select2.component.ts

```diff
@@ -95,13 +112,37 @@ export class Select2Component implements AfterViewInit, OnChanges, OnDestroy, On
         }
 
         this.element.on('select2:select select2:unselect', () => {
+            this.onChange(that.element.val());
+            this.onTouched();
             this.valueChanged.emit({
-                value: this.element.val(),
-                data: this.element.select2('data')
```

Comment:
Yes, your reasuoning is perfectly valid.
So far we haven't ran into this kind of use case.
We usually just set the id the same as the text value, so when we select an item, we basically get the object we wanted the get. Haven't used additional fields at all.
How about we go with the much simpler version now, and worry about returning more complex objects later?